larpandoracontent/LArHelpers/LArFormattingHelper.h

```diff
@@ -0,0 +1,222 @@
+ *
+ *  @brief  Header file for the formatting helper class
+ *
+ *  $Log: $
+ */
+#ifndef LAR_FORMATTING_HELPER_H
+#define LAR_FORMATTING_HELPER_H 1
+
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace lar_content
+{
+
+/**
+ *  @brief  LArFormattingHelper class
+ */
+class LArFormattingHelper
+{
+public:
+
+    enum Colour : int;
+    enum Style : int;
+
+    /**
+     *  @brief  Set the format style (to standard output stream)
+     *
+     *  @param  style the style of choice
+     */
+    static void SetStyle(const Style &style, std::ostream &stream=std::cout);
+    
+    /**
+     *  @brief  Set the text colour (of standard output stream)
+     *
+     *  @param  colour the colour of choice
+     */
+    static void SetColour(const Colour &colour, std::ostream &stream=std::cout);
+    
+    /**
+     *  @brief  Reset the style of the standard output stream
+     */
+    static void ResetStyle(std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Reset the text colour of the standard output stream
+     */
+    static void ResetColour(std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Reset the formatting and text colour of the standard output stream
+     */
+    static void Reset(std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Print a formatting character to the standard output stream
+     *
+     *  @param  code the formatting code to output
+     */
+    static void PrintFormatCharacter(const unsigned int &code, std::ostream &stream=std::cout);
+
+    /**
+     *  @brief  Get a formatting character
+     *
+     *  @param  code the formatting code to output
+     */
+    static std::string GetFormatCharacter(const unsigned int &code);
+
+    /**
+     *  @brief  Print a header line of a given width
+     *
+     *  @param  title the title of the header
+     *  @param  width the width of the header line
+     */
+    static void PrintHeader(const std::string &title = "", const unsigned int &width = 140);
+
+    /**
+     *  @brief  Print a horizontal rule
+     *
+     *  @param  width the width of the rule line
+     */
+    static void PrintRule(const unsigned int &width = 140);
+
+    enum Style : int
+    {
+        REGULAR = 0,
+        BOLD = 1,
+        DIM = 2,
+        UNDERLINED = 4,
+        INVERTED = 7
+    };
+    
+    enum Colour : int
+    {
+        DEFAULT = 39,
+        BLACK = 30,
+        RED = 31,
+        GREEN = 32,
+        YELLOW = 33,
+        BLUE = 34,
+        MAGENTA = 35,
+        CYAN = 36,
+        LIGHT_GRAY = 37,
+        DARK_GRAY = 90,
+        LIGHT_RED = 91,
+        LIGHT_GREEN = 92,
+        LIGHT_YELLOW = 93,
+        LIGHT_BLUE = 94,
+        LIGHT_MAGENTA = 95,
+        LIGHT_CYAN = 96,
+        WHITE = 97
+    };
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    class Table 
```

Comment:
Doxygen.

In Pandora code, we typically tend to add nested classes at "top" of parent class description.